src/MICore/UnixUtilities.cs

```diff
@@ -160,5 +163,28 @@ internal static bool IsProcessRunning(int processId)
             // if there is no process with the ID specified.
             return UnixNativeMethods.GetPGid(processId) >= 0;
         }
+
+        public static bool IsBinarySigned(string filePath)
```

Comment:
I believe `codesign -v` will do the hashing. I wasn't able to check whether it verifies against revocation.

I don't think that would be an issue for self-signed binaries (like we are suggesting right now in the README).